### PR TITLE
ci: add QG1 cluster readiness gate

### PR DESCRIPTION
**Pull request by kami619**
## Description

- Add a reusable `run-qg1` composite action and standalone `QG1: Cluster Readiness` workflow
- Keep cluster setup/login shared via `setup-cluster` while keeping QG1 focused on readiness assertions
- Register QG1 in the CI health dashboard and update CI alerting docs

## Jira Ticket

- https://redhat.atlassian.net/browse/RHAIENG-5866
- https://redhat.atlassian.net/browse/RHAIENG-5827

## Testing

- [x] `make test` passes (run from the affected agent directory)
- [x] Manual testing performed (describe steps below)
- [ ] No testing required (documentation/config change only)

Verification:
- `uv run pytest .github/scripts/tests -v` (`62 passed`)
- `ruff check .github/scripts/qg1_cluster_readiness.py .github/scripts/tests/test_qg1_cluster_readiness.py .github/scripts/tests/test_qg1_workflow_contract.py .github/scripts/generate_ci_health_page.py .github/scripts/tests/test_generate_ci_health_page.py`
- `ruff format --check .github/scripts/qg1_cluster_readiness.py .github/scripts/tests/test_qg1_cluster_readiness.py .github/scripts/tests/test_qg1_workflow_contract.py .github/scripts/generate_ci_health_page.py .github/scripts/tests/test_generate_ci_health_page.py`
- Python YAML parse sanity check for the new action/workflows

## Checklist

- [x] I have read [CONTRIBUTING.md](CONTRIBUTING.md)
- [x] No `.env` or secret files are included in this PR
- [x] All changes are within scope of the linked Jira ticket (if not, explain in Description)

## Review Guidance

- QG1 is intentionally a singleton pre-matrix gate, but the composite action inputs are parameterized for future cluster/profile matrixing.
- `setup-cluster` remains the shared login/bootstrap layer; `run-qg1` owns only readiness assertions.
- AIA message: Implemented with Cursor in an isolated worktree using a subagent-driven flow, with final acceptance review and verification before push.

AI-Attribution: AIA PAI Nc Hin R gpt-5.4 v1.0
AI-Interpretation: https://aiattribution.github.io/statements/AIA-PAI-Nc-Hin-R-?model=gpt-5.4

### CODE REVIEW

File: docs/ci-alert-policy.md [hunk 1]
```diff
@@ -72,8 +73,8 @@ flowchart TD
   QG9 -->|"retry from QG4"| QG4
 ```
 
-Only `QG4` and `QG7` emit alerts today. The remaining fingerprints are defined
-here for future use as more of the QG ladder becomes alert-backed.
+Only `QG1`, `QG4`, and `QG7` emit alerts today. The remaining fingerprints are
+defined here for future use as more of the QG ladder becomes alert-backed.
```
[the remark this answers is not in the document]
Comment (kami619) [reply docs/ci-alert-policy.md:77]: that is TMI (too much information)., skipping this suggestion.

**Review by shricharan-ks — APPROVED** [review]
Comprehensive review from 4 independent agents (security, code quality, workflow contract, test coverage). CI all green.

## Overall: Well-structured PR, LGTM

Strong points:
- Clean separation: composite action (reusable) + workflow (orchestration) + Python script (logic)
- `run_oc()` uses list-style subprocess args — no shell injection risk
- Action passes inputs via env vars, not inline interpolation — correct GH Actions security pattern
- All actions SHA-pinned with version comments
- Minimal permissions (`contents: read`, `actions: read` only where needed)
- Repository guard prevents fork execution
- Good error handling (CalledProcessError, TimeoutExpired both produce structured summaries)
- Innovative workflow contract tests that parse real YAML and execute the resolve step

## One actionable suggestion

**Missing test: `main()` returning 1 when oc succeeds but checks fail**

The happy path test verifies `exit_code == 0`. Error path tests verify `exit_code == 1` when `oc` raises CalledProcessError or TimeoutExpired. But there is no test for the primary failure path — oc succeeds but a check fails (e.g., missing namespace, no GPU nodes when required). This exercises `return 0 if summary["passed"] else 1` with `passed=False`.

Suggested test:
```python
def test_main_returns_1_when_cluster_not_ready(tmp_path, monkeypatch):
    responses = iter(["ok", "4.16.12", "", "ci-testing"])  # no GPU nodes
    monkeypatch.setattr(mod, "run_oc", lambda *a, **kw: next(responses))
    exit_code = mod.main([
        "--cluster-profile", "rhoai2",
        "--require-gpu", "true",
        "--required-namespaces", "ci-testing",
        "--summary-json", str(tmp_path / "s.json"),
        "--summary-md", str(tmp_path / "s.md"),
    ])
    assert exit_code == 1
```

## Minor observations (not blocking)

- `persist-credentials: false` missing on the main job checkout (present on notify-slack job) — consistent but not a security risk since no git push operations
- `parse_required_namespaces("")` edge case untested (returns `[]`, logic correct)
- Schedule cron `0 1 * * *` (daily 1 AM UTC) is reasonable

Approving — the test gap is easy to add in a follow-up and does not block merge.
